aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -208,6 +208,27 @@
    */
   public static final String CLIENT_FACTORY = "client.factory";
 
+  /**
+   * The type of {@link software.amazon.awssdk.http.SdkHttpClient} implementation used by {@link AwsClientFactory}
+   * If set, all AWS clients will use this specified HTTP client.
+   * If not set, CLIENT_HTTP_TYPE_DEFAULT will be used.
+   * For specific types supported, see CLIENT_HTTP_TYPE_* defined below.
+   */
+  public static final String CLIENT_HTTP_TYPE = "client.http.type";
+
+  /**
+   * If this is set under CLIENT_HTTP_TYPE, {@link software.amazon.awssdk.http.urlconnection.UrlConnectionHttpClient}
+   * will be used as the HTTP Client in {@link AwsClientFactory}
+   */
+  public static final String CLIENT_HTTP_TYPE_URL = "url";
```

Comment:
based on the AWSSDK class path, I think we should name this type as `urlconnection` instead of just `url`